test/liquidity-curve.test.js

```diff
@@ -161,6 +179,25 @@ describe('LiquidityCurve', function () {
       assert.deepEqual(curve1.combine(curve2).getPoints(), result)
       assert.deepEqual(curve2.combine(curve1).getPoints(), result)
     })
+
+    it('misc tests', function () {
+      const curve1 = new LiquidityCurve([[1711569, 1312840], [7503921, 6501735], [9033555, 9515658]])
+      const curve2 = new LiquidityCurve([[7647566, 8125921], [8184218, 17986167], [9460351, 18838810]])
+      const curve3 = new LiquidityCurve([[1302711, 4607848], [3449558, 6478308], [7395038, 9969362]])
+      const curve4 = new LiquidityCurve([[16576, 8209992], [271695, 12268681], [9287990, 15228960]])
+      assert.deepEqual(curve1.combine(curve2).getPoints(),
+        [[1711569, 1312840], [7503921, 6501735], [7647566, 8125921], [8184218, 17986167], [9033555, 18553648], [9460351, 18838810]])
+      assert.deepEqual(curve1.combine(curve3).getPoints(),
+        [[1302711, 4607848], [1711569, 4964069], [3449558, 6478308], [7395038, 9969362], [7503921, 9969362], [9033555, 9969362]])
+      assert.deepEqual(curve1.combine(curve4).getPoints(),
+        [[16576, 8209992], [271695, 12268681], [1711569, 12741428], [7503921, 14643204], [9033555, 15145422], [9287990, 15228960]])
+      assert.deepEqual(curve2.combine(curve3).getPoints(),
+        [[1302711, 4607848], [3449558, 6478308], [7395038, 9969362], [7647566, 9969362], [7747896, 9969362], [8184218, 17986167], [9460351, 18838810]])
+      assert.deepEqual(curve2.combine(curve4).getPoints(),
+        [[16576, 8209992], [271695, 12268681], [7647566, 14690367], [8011341, 14809803], [8184218, 17986167], [9287990, 18723647], [9460351, 18838810]])
+      assert.deepEqual(curve3.combine(curve4).getPoints(),
+        [[16576, 8209992], [271695, 12268681], [1302711, 12607189], [3449558, 13312054], [7395038, 14607455], [9287990, 15228960]])
+    })
```

Comment:
I mean, if you suppose your code has a bug, and if you created these unit test to test your code, then they would test for the buggy behavior, so make sure to avoid that (maybe you did already)